pkgs/applications/office/qownnotes/default.nix

```diff
@@ -1,29 +1,69 @@
-{ mkDerivation, lib, stdenv, fetchurl
-, qmake, qttools, qtbase, qtsvg, qtdeclarative, qtxmlpatterns, qtwebsockets
-, qtx11extras, qtwayland
+{ lib
+, stdenv
+, fetchurl
+, qmake
+, qttools
+, qtbase
+, qtdeclarative
+, qtsvg
+, qtwayland
+, qtwebsockets
+, qtx11extras
+, qtxmlpatterns
+, makeWrapper
+, wrapQtAppsHook
 }:
 
-mkDerivation rec {
+let
   pname = "qownnotes";
+  appname = "QOwnNotes";
   version = "22.12.3";
+in
+stdenv.mkDerivation {
+  inherit pname appname version;
 
   src = fetchurl {
     url = "https://download.tuxfamily.org/${pname}/src/${pname}-${version}.tar.xz";
-    # Fetch the checksum of current version with curl:
-    # curl https://download.tuxfamily.org/qownnotes/src/qownnotes-<version>.tar.xz.sha256
     sha256 = "sha256-fpI7RYOGmWwmau6tF8FPmY2/FtN9foWRX8/WgrNU6E8=";
   };
 
-  nativeBuildInputs = [ qmake qttools ];
+  dontWrapQtApps = true;
 
-  buildInputs = [ qtbase qtsvg qtdeclarative qtxmlpatterns qtwebsockets qtx11extras ]
-    ++ lib.optionals stdenv.isLinux [ qtwayland ];
+  preFixup = ''
+      wrapQtApp "$out/bin/${appname}"
+  '';
```

Comment:
Is this a negative effect of the additional symbolic link?

Reply:
@wegank No, this is needed anyways.
Does it build on darwin?

Reply:
I don't get why this is needed. QOwnNotes is clearly wrapped without `dontWrapQtApps = true;` and `preFixup`.

```
$ file result/bin/QOwnNotes 
result/bin/QOwnNotes: a /nix/store/m45cjiwspshg01wrk2pln2qndasky23r-bash-5.1-p16/bin/bash -e script, ASCII text executable
```

Also, after removing qtwayland on darwin, the build still fails with

```
/nix/store/dljzajdq589id89i0a7ypdl9bhiicwg0-hook/nix-support/setup-hook: line 115: /nix/store/9x201hgq41mmmbxnvj6qgvkkws84gg8c-qownnotes-22.12.3/bin/qownnotes: Too many levels of symbolic links
```

Reply:
Fixed that. It needs wrapQtAppsHook on darwin as well.